src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -7935,10 +7935,9 @@ void MacroAssembler::lightweight_lock(Register basic_lock, Register obj, Registe
   // Try to lock. Transition lock bits 0b01 => 0b00
   assert(oopDesc::mark_offset_in_bytes() == 0, "required to avoid lea");
   orr(mark, mark, markWord::unlocked_value);
-  if (EnableValhalla) {
-    // Mask inline_type bit such that we go to the slow path if object is an inline type
-    andr(mark, mark, ~((int) markWord::inline_type_bit_in_place));
-  }
+  // Mask inline_type bit such that we go to the slow path if object is an inline type
+  andr(mark, mark, ~((int) markWord::inline_type_bit_in_place));
```

Comment:
Since `EnableValhalla=true` by default, we were already exercising this path by default. 

Can we get into a situation where we do not use the object monitor table and the markword is inflated while we perform this check? We would get in trouble with this bit check if we can.